From 37e1ef43eb5cda31d5a68a5c6bd3c510a50c8ab8 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 26 Nov 2015 16:33:58 +0800
Subject: [PATCH 0694/5242] MLK-11911-6 mxc IPUv3: Minor improvement for
 _ipu_is_smfc_chan()

commit  f054d692c905e122fcc77cf068b296eec234999b from
https://source.codeaurora.org/external/imx/linux-imx.git

A minor improvement for _ipu_is_smfc_chan() to address the following
issue reported by Coverity:
Unsigned compared against 0 (NO_EFFECT)
unsigned_compare: This greater-than-or-equal-to-zero comparison of an
unsigned value is always true. dma_chan >= 0U.
	return ((dma_chan >= 0) && (dma_chan <= 3));

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/ipu3/ipu_prv.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/ipu3/ipu_prv.h b/drivers/mxc/ipu3/ipu_prv.h
index 0a77b580..025a7bd5 100644
--- a/drivers/mxc/ipu3/ipu_prv.h
+++ b/drivers/mxc/ipu3/ipu_prv.h
@@ -168,7 +168,7 @@ enum ipu_dmfc_type {
 
 static inline int _ipu_is_smfc_chan(uint32_t dma_chan)
 {
-	return ((dma_chan >= 0) && (dma_chan <= 3));
+	return dma_chan <= 3;
 }
 
 static inline u32 ipu_cm_read(struct ipu_soc *ipu, unsigned offset)
-- 
1.7.9.5

